core-web/libs/edit-content/src/lib/fields/dot-edit-content-category-field/components/dot-category-field-sidebar/dot-category-field-sidebar.component.ts

```diff
@@ -48,22 +54,32 @@ export class DotCategoryFieldSidebarComponent implements OnInit, OnDestroy {
 
     readonly store = inject(CategoryFieldStore);
 
+    readonly #destroyRef = inject(DestroyRef);
+
     ngOnInit(): void {
         this.store.getCategories();
+
+        this.#destroyRef.onDestroy(() => {
+            this.store.clean();
+        });
     }
 
     /**
-     * Handles the click event on an item.
-     *
-     * @param {number} index - The index of the item being clicked.
-     * @param {DotCategory} item - The item being clicked.
+     * Handles click events on items.
+     * @param {number} index - The index of the column being clicked.
+     * @param item - item clicked
      * @returns {void}
      */
     itemClicked({ index, item }: DotCategoryFieldItem): void {
         this.store.getCategories({ index, item });
     }
 
-    ngOnDestroy(): void {
-        this.store.clean();
+    /**
+     * Handles the selection (via checkbox) of displayed items.
+     * @param $event
+     * @param item
+     */
+    itemChecked({ selected, item }: { selected: string[]; item: DotCategory }) {
```

Comment:
@oidacra Is it possible to call this event directly in the html? like this: `(itemChecked)="this.store.updateSelected($event);"` as result to avoid creating this wrapper method?

Reply:
Yes. Less code... better.

Reply:
I removed too the `itemClicked` method.